src/module.ts

```diff
@@ -21,102 +17,124 @@ export default defineNuxtModule<ModuleOptions>({
-                  description: 'Define a custom Iconify API URL. Useful if you want to use a self-hosted Iconify API. Learn more: https://iconify.design/docs/api.',
-                  tags: ['@studioIcon material-symbols:api'],
-                },
-              },
-              publicApiFallback: {
-                $default: false,
-                $schema: {
-                  title: 'Public Iconify API fallback',
-                  description: 'Define if the public Iconify API should be used as fallback.',
-                  tags: ['@studioIcon material-symbols:public'],
-                },
-              },
-            },
-          },
+          nuxtIcon: schema,
         },
       })
     })
 
-    addComponent({
-      name: 'Icon',
-      global: true,
-      filePath: resolve('./runtime/Icon.vue'),
-    })
-    addComponent({
-      name: 'IconCSS',
-      global: true,
-      filePath: resolve('./runtime/IconCSS.vue'),
-    })
-
-    // Add Iconify collections & sort by longest first
-    const iconCollections = Object.keys(iconifyCollections).sort((a, b) => b.length - a.length)
+    // Bundle icons for server
+    const bundle: ServerBundleOptions | Promise<ServerBundleOptions> = (!options.serverBundle || options.provider !== 'server')
+      ? {}
+      : (options.serverBundle === 'auto')
+          ? discoverLocalCollections()
+          : options.serverBundle
     const template = addTemplate({
-      filename: 'icon-collections.mjs',
-      getContents: () => `export default ${JSON.stringify(iconCollections)}`,
+      filename: 'nuxt-icon-server-bundle.mjs',
       write: true,
+      async getContents() {
+        const { collections = [] } = await bundle
+        return [
+          `export const collections = {`,
+          ...collections.map(collection => `  '${collection}': () => import('@iconify-json/${collection}/icons.json').then(m => m.default),`),
+          `}`,
+        ].join('\n')
+      },
     })
-    // Add alias to `#icon-collections`
-    nuxt.options.alias['#icon-collections'] = template.dst
+    nuxt.options.nitro.alias ||= {}
+    nuxt.options.nitro.alias['#nuxt-icon-server-bundle'] = template.dst
+    nuxt.options.build.transpile ||= []
+    nuxt.options.build.transpile.push(template.dst)
 
+    // Devtools
     addCustomTab({
       name: 'icones',
       title: 'Icônes',
-      icon: 'i-arcticons-iconeration',
+      icon: 'https://icones.js.org/favicon.svg',
       view: {
         type: 'iframe',
         src: 'https://icones.js.org',
       },
     })
+
+    // Server-only runtime config for known CSS selectors
+    options.serverKnownCssClasses ||= []
+    const serverKnownCssClasses = options.serverKnownCssClasses || []
+    nuxt.options.runtimeConfig.nuxtIcon = {
+      serverKnownCssClasses,
+    }
+    nuxt.hook('nitro:init', async (_nitro) => {
+      _nitro.options.runtimeConfig.nuxtIcon = {
+        serverKnownCssClasses,
+      }
+    })
+    unocssIntegration(nuxt, options)
```

Comment:
Is it necessary to call `unocssIntegration` here even though it might not be installed?